rts/lra/narayana-lra/src/main/java/io/narayana/lra/provider/ParticipantStatusOctetStreamProvider.java

```diff
@@ -0,0 +1,33 @@
+package io.narayana.lra.provider;
+
+import org.eclipse.microprofile.lra.annotation.ParticipantStatus;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+@Provider
+@Produces(MediaType.APPLICATION_OCTET_STREAM)
+public class ParticipantStatusOctetStreamProvider implements MessageBodyWriter<ParticipantStatus> {
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return type.isAssignableFrom(ParticipantStatus.class);
+    }
+
+    @Override
+    public long getSize(ParticipantStatus participantStatus, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return 0;
+    }
+
+    @Override
+    public void writeTo(ParticipantStatus participantStatus, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream) throws IOException, WebApplicationException {
+        entityStream.write(participantStatus.toString().getBytes());
```

Comment:
It is better to use

``` java
participantStatus.name()
```

since it cannot be overridden whereas toString() can.

Reply:
ok, I will update this